components/server/src/authorization/authorizer.ts

```diff
@@ -299,11 +299,27 @@ export class Authorizer {
         if (!(await isFgaWritesEnabled(userID))) {
             return;
         }
-        const updates = [set(rel.organization(orgID).member.user(userID))];
+        const updates = [];
         if (role === "owner") {
-            updates.push(set(rel.organization(orgID).owner.user(userID)));
-        } else {
-            updates.push(remove(rel.organization(orgID).owner.user(userID)));
+            updates.push(
+                set(rel.organization(orgID).owner.user(userID)),
+                // TODO: owner has all permission member's permission should define in schema
+                // we should remove member but since old code was not, keep it to avoid breaking
+                set(rel.organization(orgID).member.user(userID)),
+                remove(rel.organization(orgID).collaborator.user(userID)),
+            );
+        } else if (role === "member") {
+            updates.push(
+                remove(rel.organization(orgID).owner.user(userID)),
+                set(rel.organization(orgID).member.user(userID)),
+                remove(rel.organization(orgID).collaborator.user(userID)),
```

Comment:
I believe this should also set the `collaborator` role. It seems like we model the roles as inclusive at least the removeOrgRole method below suggests that? I.e. if would removing the `member` role be a downgrade to `collaborator`?